department-of-veterans-affairs/caseflow
# CASEFLOW-108 #resolve Create CavcRemandProcessedLetterResponseWindowTask

Resolves #15309 (aka [CASEFLOW-108](https://vajira.max.gov/browse/CASEFLOW-108))

### Description
Create new CavcRemandProcessedLetterResponseWindowTask for 90-day duration with ability to "end hold early".

Also included the action to place task back on hold -- see  [Slack](https://dsva.slack.com/archives/C01BKL4CCRL/p1605913943002800?thread_ts=1605913086.002300&cid=C01BKL4CCRL).

### Acceptance Criteria
- [ ] Verify when a SendCavcRemandProcessedLetterTask is completed, a CavcRemandProcessedLetterResponseWindowTask is created
   - [ ] As a child of the CavcTask
   - [ ] Assigned to the CavcLitigationSupportTeam
   - [ ] Is put on hold for 90 days by creating a child timed hold task
   - [ ] Give task action of "End hold early" while task is on hold
   - [ ] After 90 days, the task should come off hold and show up in the CAVC teams unassigned tab to be assigned/acted upon

### Testing Plan
Log in as CAVC_LIT_SUPPORT_ADMIN (VACO)
1. Go to the team view of Queue: http://localhost:3000/organizations/cavc-lit-support?tab=unassignedTab&page=1
1. Click on a case in Unassigned tab; note the case docket number
1. "Assign to person" and check that case is now under the "Assigned tab" in the team view (note the task type)
1. Check task tree in the Rails console

Log in as the assignee (in an incognito window)
1.  go to their queue: http://localhost:3000/queue?tab=assigned_person&page=1
1. Click on the assigned case and "Mark task complete"; not that case is in the Completed tab in the user's queue
1. Click on the case, note the user as the action to "End hold early". (You can test it now or later.)
1. Check task tree in the Rails console

Log in as CAVC_LIT_SUPPORT_ADMIN (VACO)
1. check that case is still under the "Assigned tab" in the team view (with the Window Task type)
1. Click on the case, note the user as the action to "End hold early". (You can test it now or later.)
1. Check task tree in the Rails console

Ending and restoring hold:
1. When you "End hold early", the case shows in the Unassigned tab  in the team view.
1. When you "Put task on hold", the case shows in the "Assigned tab" in the team view.
1. Check task tree in the Rails console

Hold ends due to time passing:
1. Travel 90+ days into the future
   ```ruby
   Timecop.travel(Time.zone.now + 90.days + 1.hour)
   TaskTimerJob.perform_now
   ```
1. The case shows in the Unassigned tab  in the team view.
1. Check task tree in the Rails console

### User Facing Changes

Note window task instructions and "End hold early" action:
![image](https://user-images.githubusercontent.com/55255674/99857353-ad4fd380-2b50-11eb-88fe-1c533ff10add.png)
Case Timeline shows completed send letter task:
![image](https://user-images.githubusercontent.com/55255674/99857432-da03eb00-2b50-11eb-8292-6eaaf3ffc90e.png)

After clicking on "End hold early" action:
![image](https://user-images.githubusercontent.com/55255674/99857496-facc4080-2b50-11eb-963d-68e3db0bc31f.png)
and after clicking Submit, user can "Put task on hold" (do we want this action to be available?):
![image](https://user-images.githubusercontent.com/55255674/99857586-251dfe00-2b51-11eb-8895-b7f12a4e1ef2.png)



### Code Documentation Updates
